gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/tasks/LaunchConfigTask.java

```diff
@@ -74,6 +74,11 @@ public abstract class LaunchConfigTask extends DefaultTask {
             // AWS-managed systems that modify DNS records on failover.
             "-Dsun.net.inetaddr.ttl=20",
             "-XX:NativeMemoryTracking=summary",
+            // Improve the overall fidelity of JFR profiles and reduce sampling bias,
+            // see http://hirt.se/blog/?p=609 & https://jpbempel.github.io/2022/06/22/debug-non-safepoints.html
+            // Per https://bugs.openjdk.org/browse/JDK-8201516 may still be potentially incorrect attribution.
+            "-XX:+UnlockDiagnosticVMOptions",
+            "-XX:+DebugNonSafepoints",
```

Comment:
Consider testing impact of preserving frame pointers for native and kernel traces, though that may be too expensive to enable by default:

* https://github.com/jvm-profiling-tools/async-profiler
* http://techblog.netflix.com/2015/07/java-in-flames.html
* https://bugs.openjdk.org/browse/JDK-8068945


```suggestion
            "-XX:+DebugNonSafepoints",
            "-XX:+ PreserveFramePointer",
```

Reply:
I haven't had a chance to benchmark myself, do you know if there's a measurable runtime cost associated with `+ DebugNonSafepoints`?

Reply:
I haven't had cycles yet to benchmark with/without `-XX:+DebugNonSafepoints` on current JDKs  for our various workflows such as WC infra. The JDK ticket and [Jean-Phillipe Bempel's `Why JVM modern profilers are still safepoint biased?` blog post](https://jpbempel.github.io/2022/06/22/debug-non-safepoints.html) show ~2% overhead. If you have some pointers to spin those WC benchmarks up, I can try to give them a shot to understand potential fleet implications.

Async-profiler enables DebugNonSafepoints via JVMTI agent registering [CompileMethodLoad](https://github.com/jvm-profiling-tools/async-profiler/blob/master/src/profiler.h#L222-L228)

Also worth reading the note http://hirt.se/blog/?p=609 from Marcus Hirt, one of the original JRockit and JFR authors.

Reply:
Benchmarks don't appear to be impacted (results are within margin of error)